### PR TITLE
Integers set

### PR DESCRIPTION
**Pull request by rikardn**
Add the set of integers and fix set_complement for Reals

### CODE REVIEW

File: symengine/sets.cpp [hunk 1]
```diff
@@ -154,6 +156,30 @@ RCP<const Set> Interval::set_intersection(const RCP<const Set> &o) const
...
+                return emptyset();
+            }
+            set_basic container;
+            while (eq(*Ge(last, first), *boolTrue)) {
+                container.insert(first);
+                first = add(first, integer(1));
+            }
+            return finiteset(container);
+        } else {
+            return SymEngine::set_intersection(
```
Comment (isuruf) [inline symengine/sets.cpp:179]: For an interval like,
`(x, y]` intersecting with Integers should give, `[floor(x) + 1, floor(y)]`
`(x, y)` intersecting with Integers should give, `[floor(x) + 1, ceiling(y)-1]`
`[x, y]` intersecting with Integers should give, `[celing(x), floor(y)]`
`[x, y)` intersecting with Integers should give, `[ceiling(x), ceiling(y)-1]`

Comment (isuruf) [reply symengine/sets.cpp:179]: nvm, ignore the above

**Comment by isuruf** [comment]
Looks good to me. Thanks for the PR

**Comment by rikardn** [comment]
Thanks!

**Comment by certik** [comment]
Thanks @rikardn for the PR and thanks @isuruf for the review.

**Pull request merged**